### PR TITLE
Documentation comments

### PR DESCRIPTION
**Pull request by carson-katri**
This is mainly to support my WIP Tokamak site [TokamakDocs](https://github.com/carson-katri/TokamakDocs). It will auto generate the site from the doc comments on Views in TokamakCore.

I want to make sure there isn't any "No overview available." before merging.

### CODE REVIEW

**Comment by carson-katri** [comment]
Here's an example of the generated docs:
Screen Shot 2020-07-01 at 7 00 08 PM

All example code blocks in doc comments are translated to demo views, rendered by Tokamak.

File: Sources/TokamakCore/Views/TextField.swift [hunk 1]
```diff
@@ -15,6 +15,26 @@
 //  Created by Jed Fox on 06/28/2020.
 //
 
+/// A control that displays an editable text interface.
+///
+/// Available when `Label` conforms to `View`
+/// The
```
Comment (j-f1) [inline Sources/TokamakCore/Views/TextField.swift:21]: Missing a sentence here?

**Comment by MaxDesiatov** [comment]
Oh, I've realised just now that this provides inline previews for example code 🤯

Fantastic work @carson-katri! 👏

File: Sources/TokamakCore/Views/ForEach.swift [hunk 2]
```diff
@@ -12,6 +12,18 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/// A structure that computes views on demand from an underlying collection of of identified data.
+///
+/// Available when `Data` conforms to `RandomAccessCollection`, `ID` conforms to `Hashable`, and `Content` conforms to `View`.
```
Comment (ie-ahm-robox) [inline Sources/TokamakCore/Views/ForEach.swift:17]: - :no_entry_sign: Line should be 100 characters or less: currently 126 characters (`line_length`)

**Comment by ie-ahm-robox** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/TokamakCore/Views/AnyView.swift#L27** - Line should be 100 characters or less: currently 101 characters (`line_length`)
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/TokamakCore/Views/ForEach.swift#L27** - Line should be 100 characters or less: currently 112 characters (`line_length`)
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/TokamakCore/Views/ScrollView.swift#L29** - Line should be 100 characters or less: currently 118 characters (`line_length`)
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/TokamakCore/Views/ScrollView.swift#L43** - Line should be 100 characters or less: currently 112 characters (`line_length`)
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 3893b2a06958f38c9de405bae362c1e00f208e19
</p>

File: Sources/TokamakCore/Views/ForEach.swift [hunk 3]
```diff
@@ -12,6 +12,18 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/// A structure that computes views on demand from an underlying collection of of identified data.
```
Comment (MaxDesiatov) [inline Sources/TokamakCore/Views/ForEach.swift:15]:
```suggestion
/// A structure that computes views on demand from an underlying collection of identified data.
```

File: Sources/TokamakCore/Views/ForEach.swift [hunk 4]
```diff
@@ -12,6 +12,19 @@
...
+///
+/// The children computed by `ForEach` are directly passed to the encapsulating `View`.
+/// Similar to `TupleView` and `Group`.
+///
+///     HStack {
+///       ForEach(0..<5) {
+///         Text("\($0)")
+///       }
+///     }
 public struct ForEach<Data, ID, Content>: View where Data: RandomAccessCollection, ID: Hashable, Content: View {
```
Comment (ie-ahm-robox) [inline Sources/TokamakCore/Views/ForEach.swift:28]: - :no_entry_sign: Line should be 100 characters or less: currently 112 characters (`line_length`)

**Comment by ie-ahm-robox** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/TokamakCore/Views/AnyView.swift#L27** - Line should be 100 characters or less: currently 101 characters (`line_length`)
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/TokamakCore/Views/ScrollView.swift#L29** - Line should be 100 characters or less: currently 118 characters (`line_length`)
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **Sources/TokamakCore/Views/ScrollView.swift#L43** - Line should be 100 characters or less: currently 112 characters (`line_length`)
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 4fd0f63235289d5c24b5de61539d38afca3b6179
</p>

File: Sources/TokamakCore/Views/Button.swift [hunk 5]
```diff
@@ -15,6 +15,25 @@
 //  Created by Max Desiatov on 02/12/2018.
 //
 
+/// A control that performs an action when triggered.
+///
+/// Available when `Label` conforms to `View`.
+///
+/// A button is created using a `Label` an action (a method or closure) to be called on click.
```
Comment (MaxDesiatov) [inline Sources/TokamakCore/Views/Button.swift:22]: This doesn't read smoothly to me, how's about slightly changing it similar to this?
```suggestion
/// A button is created using a `Label`, the `action` initializer argument (a method or closure) 
/// is called on click.
```

File: Sources/TokamakCore/Views/ForEach.swift (hunk 4 above)
Comment (ie-ahm-robox) [inline Sources/TokamakCore/Views/ForEach.swift:28]: - :no_entry_sign: Line should be 100 characters or less: currently 112 characters (`line_length`)

Comment (ie-ahm-robox) [inline Sources/TokamakCore/Views/ForEach.swift:28]: - :no_entry_sign: Line should be 100 characters or less: currently 112 characters (`line_length`)

Comment (MaxDesiatov) [reply Sources/TokamakCore/Views/ForEach.swift:28]: @carson-katri could you also fix this one please?

Comment (MaxDesiatov) [inline Sources/TokamakCore/Views/ForEach.swift:28]:
```suggestion
public struct ForEach<Data, ID, Content>: View
  where Data: RandomAccessCollection, ID: Hashable, Content: View {
```

**Pull request merged**